{"matches":[{"vulnerability":{"id":"GHSA-j8r2-6x86-q33q","dataSource":"https://github.com/advisories/GHSA-j8r2-6x86-q33q","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Unintended leak of Proxy-Authorization header in requests","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:R/S:C/C:H/I:N/A:N","metrics":{"baseScore":6.1,"exploitabilityScore":1.7,"impactScore":4},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-32681","epss":0.04731,"percentile":0.88915,"date":"2025-06-02"}],"fix":{"versions":["2.31.0"],"state":"fixed"},"advisories":[],"risk":2.6257049999999995},"relatedVulnerabilities":[{"id":"CVE-2023-32681","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-32681","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/psf/requests/commit/74ea7cf7a6a27a4eeb2ae24e162bcc942a6706d5","https://github.com/psf/requests/releases/tag/v2.31.0","https://github.com/psf/requests/security/advisories/GHSA-j8r2-6x86-q33q","https://lists.debian.org/debian-lts-announce/2023/06/msg00018.html","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/AW7HNFGYP44RT3DUDQXG2QT3OEV2PJ7Y/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KOYASTZDGQG2BWLSNBPL3TQRL2G7QYNZ/","https://security.gentoo.org/glsa/202309-08","https://github.com/psf/requests/commit/74ea7cf7a6a27a4eeb2ae24e162bcc942a6706d5","https://github.com/psf/requests/releases/tag/v2.31.0","https://github.com/psf/requests/security/advisories/GHSA-j8r2-6x86-q33q","https://lists.debian.org/debian-lts-announce/2023/06/msg00018.html","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/AW7HNFGYP44RT3DUDQXG2QT3OEV2PJ7Y/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KOYASTZDGQG2BWLSNBPL3TQRL2G7QYNZ/","https://security.gentoo.org/glsa/202309-08"],"description":"Requests is a HTTP library. Since Requests 2.3.0, Requests has been leaking Proxy-Authorization headers to destination servers when redirected to an HTTPS endpoint. This is a product of how we use `rebuild_proxies` to reattach the `Proxy-Authorization` header to requests. For HTTP connections sent through the tunnel, the proxy will identify the header in the request itself and remove it prior to forwarding to the destination server. However when sent over HTTPS, the `Proxy-Authorization` header must be sent in the CONNECT request as the proxy has no visibility into the tunneled request. This results in Requests forwarding proxy credentials to the destination server unintentionally, allowing a malicious actor to potentially exfiltrate sensitive information. This issue has been patched in version 2.31.0.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:R/S:C/C:H/I:N/A:N","metrics":{"baseScore":6.1,"exploitabilityScore":1.7,"impactScore":4},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:R/S:C/C:H/I:N/A:N","metrics":{"baseScore":6.1,"exploitabilityScore":1.7,"impactScore":4},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-32681","epss":0.04731,"percentile":0.88915,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"requests","version":"2.25.1"}},"found":{"versionConstraint":">=2.3.0,<2.31.0 (python)","vulnerabilityID":"GHSA-j8r2-6x86-q33q"},"fix":{"suggestedVersion":"2.31.0"}}],"artifact":{"id":"52e214e010c57a74","name":"requests","version":"2.25.1","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/requests-2.25.1.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/requests-2.25.1.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/requests-2.25.1.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/requests-2.25.1.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["Apache 2.0"],"cpes":["cpe:2.3:a:python:requests:2.25.1:*:*:*:*:*:*:*"],"purl":"pkg:pypi/requests@2.25.1","upstreams":[]}},{"vulnerability":{"id":"GHSA-q34m-jh98-gwm2","dataSource":"https://github.com/advisories/GHSA-q34m-jh98-gwm2","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Werkzeug possible resource exhaustion when parsing file data in forms","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-49767","epss":0.00974,"percentile":0.75666,"date":"2025-06-02"}],"fix":{"versions":["3.0.6"],"state":"fixed"},"advisories":[],"risk":0.60875},"relatedVulnerabilities":[{"id":"CVE-2024-49767","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-49767","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pallets/quart/commit/5e78c4169b8eb66b91ead3e62d44721b9e1644ee","https://github.com/pallets/quart/commit/abb04a512496206de279225340ed022852fbf51f","https://github.com/pallets/werkzeug/commit/50cfeebcb0727e18cc52ffbeb125f4a66551179b","https://github.com/pallets/werkzeug/releases/tag/3.0.6","https://github.com/pallets/werkzeug/security/advisories/GHSA-q34m-jh98-gwm2","https://security.netapp.com/advisory/ntap-20250103-0007/"],"description":"Werkzeug is a Web Server Gateway Interface web application library. Applications using `werkzeug.formparser.MultiPartParser` corresponding to a version of Werkzeug prior to 3.0.6 to parse `multipart/form-data` requests (e.g. all flask applications) are vulnerable to a relatively simple but effective resource exhaustion (denial of service) attack. A specifically crafted form submission request can cause the parser to allocate and block 3 to 8 times the upload size in main memory. There is no upper limit; a single upload at 1 Gbit/s can exhaust 32 GB of RAM in less than 60 seconds. Werkzeug version 3.0.6 fixes this issue.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:N/AC:L/AT:N/PR:N/UI:N/VC:N/VI:N/VA:L/SC:N/SI:N/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":6.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-49767","epss":0.00974,"percentile":0.75666,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"werkzeug","version":"2.0.2"}},"found":{"versionConstraint":"<=3.0.5 (python)","vulnerabilityID":"GHSA-q34m-jh98-gwm2"},"fix":{"suggestedVersion":"3.0.6"}}],"artifact":{"id":"25eafd70a0c695cf","name":"werkzeug","version":"2.0.2","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/Werkzeug-2.0.2.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/Werkzeug-2.0.2.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/Werkzeug-2.0.2.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/Werkzeug-2.0.2.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:armin_ronacher_project:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python:werkzeug:2.0.2:*:*:*:*:*:*:*"],"purl":"pkg:pypi/werkzeug@2.0.2","upstreams":[]}},{"vulnerability":{"id":"GHSA-v845-jxx5-vc9f","dataSource":"https://github.com/advisories/GHSA-v845-jxx5-vc9f","namespace":"github:language:python","severity":"High","urls":[],"description":"`Cookie` HTTP header isn't stripped on cross-origin redirects","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:H/UI:N/S:U/C:H/I:H/A:N","metrics":{"baseScore":5.9,"exploitabilityScore":0.8,"impactScore":5.2},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-43804","epss":0.00554,"percentile":0.67074,"date":"2025-06-02"}],"fix":{"versions":["1.26.17"],"state":"fixed"},"advisories":[],"risk":0.37118},"relatedVulnerabilities":[{"id":"CVE-2023-43804","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-43804","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/urllib3/urllib3/commit/01220354d389cd05474713f8c982d05c9b17aafb","https://github.com/urllib3/urllib3/commit/644124ecd0b6e417c527191f866daa05a5a2056d","https://github.com/urllib3/urllib3/security/advisories/GHSA-v845-jxx5-vc9f","https://lists.debian.org/debian-lts-announce/2023/10/msg00012.html","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5F5CUBAN5XMEBVBZPHFITBLMJV5FIJJ5/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/I3PR7C6RJ6JUBQKIJ644DMIJSUP36VDY/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/NDAGZXYJ7H2G3SB47M453VQVNAWKAEJJ/","https://github.com/urllib3/urllib3/commit/01220354d389cd05474713f8c982d05c9b17aafb","https://github.com/urllib3/urllib3/commit/644124ecd0b6e417c527191f866daa05a5a2056d","https://github.com/urllib3/urllib3/security/advisories/GHSA-v845-jxx5-vc9f","https://lists.debian.org/debian-lts-announce/2023/10/msg00012.html","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5F5CUBAN5XMEBVBZPHFITBLMJV5FIJJ5/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/I3PR7C6RJ6JUBQKIJ644DMIJSUP36VDY/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/NDAGZXYJ7H2G3SB47M453VQVNAWKAEJJ/","https://security.netapp.com/advisory/ntap-20241213-0007/","https://www.vicarius.io/vsociety/posts/cve-2023-43804-urllib3-vulnerability-3"],"description":"urllib3 is a user-friendly HTTP client library for Python. urllib3 doesn't treat the `Cookie` HTTP header special or provide any helpers for managing cookies over HTTP, that is the responsibility of the user. However, it is possible for a user to specify a `Cookie` header and unknowingly leak information via HTTP redirects to a different origin if that user doesn't disable redirects explicitly. This issue has been patched in urllib3 version 1.26.17 or 2.0.5.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:N","metrics":{"baseScore":8.1,"exploitabilityScore":2.9,"impactScore":5.2},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:H/UI:N/S:U/C:H/I:H/A:N","metrics":{"baseScore":5.9,"exploitabilityScore":0.8,"impactScore":5.2},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-43804","epss":0.00554,"percentile":0.67074,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"urllib3","version":"1.26.5"}},"found":{"versionConstraint":"<1.26.17 (python)","vulnerabilityID":"GHSA-v845-jxx5-vc9f"},"fix":{"suggestedVersion":"1.26.17"}}],"artifact":{"id":"043d634961733646","name":"urllib3","version":"1.26.5","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/urllib3-1.26.5.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/urllib3-1.26.5.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/urllib3-1.26.5.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/urllib3-1.26.5.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:python:urllib3:1.26.5:*:*:*:*:*:*:*"],"purl":"pkg:pypi/urllib3@1.26.5","upstreams":[]}},{"vulnerability":{"id":"GHSA-5rjg-fvgr-3xxf","dataSource":"https://github.com/advisories/GHSA-5rjg-fvgr-3xxf","namespace":"github:language:python","severity":"High","urls":[],"description":"setuptools has a path traversal vulnerability in PackageIndex.download that leads to Arbitrary File Write","cvss":[],"epss":[{"cve":"CVE-2025-47273","epss":0.00435,"percentile":0.6206,"date":"2025-06-02"}],"fix":{"versions":["78.1.1"],"state":"fixed"},"advisories":[],"risk":0.32625},"relatedVulnerabilities":[{"id":"CVE-2025-47273","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2025-47273","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pypa/setuptools/blob/6ead555c5fb29bc57fe6105b1bffc163f56fd558/setuptools/package_index.py#L810C1-L825C88","https://github.com/pypa/setuptools/commit/250a6d17978f9f6ac3ac887091f2d32886fbbb0b","https://github.com/pypa/setuptools/issues/4946","https://github.com/pypa/setuptools/security/advisories/GHSA-5rjg-fvgr-3xxf","https://lists.debian.org/debian-lts-announce/2025/05/msg00035.html","https://github.com/pypa/setuptools/issues/4946"],"description":"setuptools is a package that allows users to download, build, install, upgrade, and uninstall Python packages. A path traversal vulnerability in `PackageIndex` is present in setuptools prior to version 78.1.1. An attacker would be allowed to write files to arbitrary locations on the filesystem with the permissions of the process running the Python code, which could escalate to remote code execution depending on the context. Version 78.1.1 fixes the issue.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:N/AC:L/AT:N/PR:N/UI:N/VC:N/VI:H/VA:N/SC:N/SI:N/SA:N/E:P/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":7.7},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2025-47273","epss":0.00435,"percentile":0.6206,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"setuptools","version":"59.6.0"}},"found":{"versionConstraint":"<78.1.1 (python)","vulnerabilityID":"GHSA-5rjg-fvgr-3xxf"},"fix":{"suggestedVersion":"78.1.1"}}],"artifact":{"id":"904f7c7f07290780","name":"setuptools","version":"59.6.0","type":"python","locations":[{"path":"/lib/python3.10/site-packages/setuptools-59.6.0.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/setuptools-59.6.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/setuptools-59.6.0.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/setuptools-59.6.0.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/setuptools-59.6.0.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/setuptools-59.6.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["UNKNOWN"],"cpes":["cpe:2.3:a:python:setuptools:59.6.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/setuptools@59.6.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-wrxv-2j5q-m38w","dataSource":"https://github.com/advisories/GHSA-wrxv-2j5q-m38w","namespace":"github:language:python","severity":"Medium","urls":[],"description":"lxml NULL Pointer Dereference allows attackers to cause a denial of service","cvss":[{"type":"Secondary","version":"3.0","vector":"CVSS:3.0/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:L","metrics":{"baseScore":5.3,"exploitabilityScore":3.9,"impactScore":1.5},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-2309","epss":0.00513,"percentile":0.65602,"date":"2025-06-02"}],"fix":{"versions":["4.9.1"],"state":"fixed"},"advisories":[],"risk":0.264195},"relatedVulnerabilities":[{"id":"CVE-2022-2309","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2022-2309","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/lxml/lxml/commit/86368e9cf70a0ad23cccd5ee32de847149af0c6f","https://huntr.dev/bounties/8264e74f-edda-4c40-9956-49de635105ba","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/HGYC6L7ENH5VEGN3YWFBYMGKX6WNS7HZ/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/URHHSIBTPTALXMECRLAC2EVDNAFSR5NO/","https://security.gentoo.org/glsa/202208-06","https://security.netapp.com/advisory/ntap-20220915-0006/","https://github.com/lxml/lxml/commit/86368e9cf70a0ad23cccd5ee32de847149af0c6f","https://huntr.dev/bounties/8264e74f-edda-4c40-9956-49de635105ba","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/HGYC6L7ENH5VEGN3YWFBYMGKX6WNS7HZ/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/URHHSIBTPTALXMECRLAC2EVDNAFSR5NO/","https://security.gentoo.org/glsa/202208-06","https://security.netapp.com/advisory/ntap-20220915-0006/"],"description":"NULL Pointer Dereference allows attackers to cause a denial of service (or application crash). This only applies when lxml is used together with libxml2 2.9.10 through 2.9.14. libxml2 2.9.9 and earlier are not affected. It allows triggering crashes through forged input data, given a vulnerable code sequence in the application. The vulnerability is caused by the iterwalk function (also used by the canonicalize function). Such code shouldn't be in wide-spread use, given that parsing + iterwalk would usually be replaced with the more efficient iterparse function. However, an XML converter that serialises to C14N would also be vulnerable, for example, and there are legitimate use cases for this code sequence. If untrusted input is received (also remotely) and processed via iterwalk function, a crash can be triggered.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"nvd@nist.gov","type":"Primary","version":"2.0","vector":"AV:N/AC:L/Au:N/C:N/I:N/A:P","metrics":{"baseScore":5,"exploitabilityScore":10,"impactScore":2.9},"vendorMetadata":{}},{"source":"security@huntr.dev","type":"Secondary","version":"3.0","vector":"CVSS:3.0/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:L","metrics":{"baseScore":5.3,"exploitabilityScore":3.9,"impactScore":1.5},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-2309","epss":0.00513,"percentile":0.65602,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"lxml","version":"4.8.0"}},"found":{"versionConstraint":"<4.9.1 (python)","vulnerabilityID":"GHSA-wrxv-2j5q-m38w"},"fix":{"suggestedVersion":"4.9.1"}}],"artifact":{"id":"6e710e695023b6d5","name":"lxml","version":"4.8.0","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/lxml-4.8.0.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/lxml-4.8.0.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/lxml-4.8.0.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/lxml-4.8.0.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD"],"cpes":["cpe:2.3:a:lxml:lxml:4.8.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/lxml@4.8.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-jjg7-2v4v-x38h","dataSource":"https://github.com/advisories/GHSA-jjg7-2v4v-x38h","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Internationalized Domain Names in Applications (IDNA) vulnerable to denial of service from specially crafted inputs to idna.encode","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":6.2,"exploitabilityScore":2.6,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-3651","epss":0.00436,"percentile":0.62112,"date":"2025-06-02"}],"fix":{"versions":["3.7"],"state":"fixed"},"advisories":[],"risk":0.24416000000000004},"relatedVulnerabilities":[{"id":"CVE-2024-3651","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-3651","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/kjd/idna/commit/1d365e17e10d72d0b7876316fc7b9ca0eebdd38d","https://huntr.com/bounties/93d78d07-d791-4b39-a845-cbfabc44aadb","https://github.com/kjd/idna/commit/1d365e17e10d72d0b7876316fc7b9ca0eebdd38d","https://huntr.com/bounties/93d78d07-d791-4b39-a845-cbfabc44aadb"],"description":"A vulnerability was identified in the kjd/idna library, specifically within the `idna.encode()` function, affecting version 3.6. The issue arises from the function's handling of crafted input strings, which can lead to quadratic complexity and consequently, a denial of service condition. This vulnerability is triggered by a crafted input that causes the `idna.encode()` function to process the input with considerable computational load, significantly increasing the processing time in a quadratic manner relative to the input size.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"security@huntr.dev","type":"Secondary","version":"3.0","vector":"CVSS:3.0/AV:L/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":6.2,"exploitabilityScore":2.6,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-3651","epss":0.00436,"percentile":0.62112,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"idna","version":"3.3"}},"found":{"versionConstraint":"<3.7 (python)","vulnerabilityID":"GHSA-jjg7-2v4v-x38h"},"fix":{"suggestedVersion":"3.7"}}],"artifact":{"id":"30be7e210695bf55","name":"idna","version":"3.3","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/idna-3.3.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/idna-3.3.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/idna-3.3.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/idna-3.3.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:kim_davies_project:python-idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim_davies_project:python_idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim_daviesproject:python-idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim_daviesproject:python_idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim_davies_project:idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim_project:python-idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim_project:python_idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:python-idna:python-idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:python-idna:python_idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:python_idna:python-idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:python_idna:python_idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim_davies:python-idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim_davies:python_idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim_daviesproject:idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kimproject:python-idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kimproject:python_idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:python:python-idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:python:python_idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:idna:python-idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:idna:python_idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim_project:idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:python-idna:idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:python_idna:idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim:python-idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim:python_idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim_davies:idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kimproject:idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:python:idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:idna:idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim:idna:3.3:*:*:*:*:*:*:*"],"purl":"pkg:pypi/idna@3.3","upstreams":[]}},{"vulnerability":{"id":"GHSA-r9hx-vwmv-q579","dataSource":"https://github.com/advisories/GHSA-r9hx-vwmv-q579","namespace":"github:language:python","severity":"High","urls":[],"description":"pypa/setuptools vulnerable to Regular Expression Denial of Service (ReDoS)","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-40897","epss":0.00318,"percentile":0.54322,"date":"2025-06-02"}],"fix":{"versions":["65.5.1"],"state":"fixed"},"advisories":[],"risk":0.2385},"relatedVulnerabilities":[{"id":"CVE-2022-40897","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2022-40897","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/pypa/setuptools/blob/fe8a98e696241487ba6ac9f91faa38ade939ec5d/setuptools/package_index.py#L200","https://github.com/pypa/setuptools/commit/43a9c9bfa6aa626ec2a22540bea28d2ca77964be","https://github.com/pypa/setuptools/compare/v65.5.0...v65.5.1","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/ADES3NLOE5QJKBLGNZNI2RGVOSQXA37R/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/YNA2BAH2ACBZ4TVJZKFLCR7L23BG5C3H/","https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/","https://pyup.io/vulnerabilities/CVE-2022-40897/52495/","https://security.netapp.com/advisory/ntap-20230214-0001/","https://security.netapp.com/advisory/ntap-20240621-0006/","https://github.com/pypa/setuptools/blob/fe8a98e696241487ba6ac9f91faa38ade939ec5d/setuptools/package_index.py#L200","https://github.com/pypa/setuptools/commit/43a9c9bfa6aa626ec2a22540bea28d2ca77964be","https://github.com/pypa/setuptools/compare/v65.5.0...v65.5.1","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/ADES3NLOE5QJKBLGNZNI2RGVOSQXA37R/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/YNA2BAH2ACBZ4TVJZKFLCR7L23BG5C3H/","https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/","https://pyup.io/vulnerabilities/CVE-2022-40897/52495/","https://security.netapp.com/advisory/ntap-20230214-0001/","https://security.netapp.com/advisory/ntap-20240621-0006/"],"description":"Python Packaging Authority (PyPA) setuptools before 65.5.1 allows remote attackers to cause a denial of service via HTML in a crafted package or custom PackageIndex page. There is a Regular Expression Denial of Service (ReDoS) in package_index.py.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":5.9,"exploitabilityScore":2.3,"impactScore":3.6},"vendorMetadata":{}},{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":5.9,"exploitabilityScore":2.3,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-40897","epss":0.00318,"percentile":0.54322,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"setuptools","version":"59.6.0"}},"found":{"versionConstraint":"<65.5.1 (python)","vulnerabilityID":"GHSA-r9hx-vwmv-q579"},"fix":{"suggestedVersion":"65.5.1"}}],"artifact":{"id":"904f7c7f07290780","name":"setuptools","version":"59.6.0","type":"python","locations":[{"path":"/lib/python3.10/site-packages/setuptools-59.6.0.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/setuptools-59.6.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/setuptools-59.6.0.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/setuptools-59.6.0.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/setuptools-59.6.0.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/setuptools-59.6.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["UNKNOWN"],"cpes":["cpe:2.3:a:python:setuptools:59.6.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/setuptools@59.6.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-xg9f-g7g7-2323","dataSource":"https://github.com/advisories/GHSA-xg9f-g7g7-2323","namespace":"github:language:python","severity":"High","urls":[],"description":"High resource usage when parsing multipart form data with many fields","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-25577","epss":0.00306,"percentile":0.53406,"date":"2025-06-02"}],"fix":{"versions":["2.2.3"],"state":"fixed"},"advisories":[],"risk":0.22949999999999998},"relatedVulnerabilities":[{"id":"CVE-2023-25577","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-25577","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pallets/werkzeug/commit/517cac5a804e8c4dc4ed038bb20dacd038e7a9f1","https://github.com/pallets/werkzeug/releases/tag/2.2.3","https://github.com/pallets/werkzeug/security/advisories/GHSA-xg9f-g7g7-2323","https://security.netapp.com/advisory/ntap-20230818-0003/","https://www.debian.org/security/2023/dsa-5470","https://github.com/pallets/werkzeug/commit/517cac5a804e8c4dc4ed038bb20dacd038e7a9f1","https://github.com/pallets/werkzeug/releases/tag/2.2.3","https://github.com/pallets/werkzeug/security/advisories/GHSA-xg9f-g7g7-2323","https://security.netapp.com/advisory/ntap-20230818-0003/","https://www.debian.org/security/2023/dsa-5470"],"description":"Werkzeug is a comprehensive WSGI web application library. Prior to version 2.2.3, Werkzeug's multipart form data parser will parse an unlimited number of parts, including file parts. Parts can be a small amount of bytes, but each requires CPU time to parse and may use more memory as Python data. If a request can be made to an endpoint that accesses `request.data`, `request.form`, `request.files`, or `request.get_data(parse_form_data=False)`, it can cause unexpectedly high resource usage. This allows an attacker to cause a denial of service by sending crafted multipart data to an endpoint that will parse it. The amount of CPU time required can block worker processes from handling legitimate requests. The amount of RAM required can trigger an out of memory kill of the process. Unlimited file parts can use up memory and file handles. If many concurrent requests are sent continuously, this can exhaust or kill all available workers. Version 2.2.3 contains a patch for this issue.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-25577","epss":0.00306,"percentile":0.53406,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"werkzeug","version":"2.0.2"}},"found":{"versionConstraint":"<2.2.3 (python)","vulnerabilityID":"GHSA-xg9f-g7g7-2323"},"fix":{"suggestedVersion":"2.2.3"}}],"artifact":{"id":"25eafd70a0c695cf","name":"werkzeug","version":"2.0.2","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/Werkzeug-2.0.2.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/Werkzeug-2.0.2.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/Werkzeug-2.0.2.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/Werkzeug-2.0.2.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:armin_ronacher_project:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python:werkzeug:2.0.2:*:*:*:*:*:*:*"],"purl":"pkg:pypi/werkzeug@2.0.2","upstreams":[]}},{"vulnerability":{"id":"GHSA-hrfv-mqp8-q5rw","dataSource":"https://github.com/advisories/GHSA-hrfv-mqp8-q5rw","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Werkzeug DoS: High resource usage when parsing multipart/form-data containing a large part with CR/LF character at the beginning","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:A/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":5.7,"exploitabilityScore":2.1,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-46136","epss":0.0041,"percentile":0.60519,"date":"2025-06-02"}],"fix":{"versions":["2.3.8"],"state":"fixed"},"advisories":[],"risk":0.21935000000000002},"relatedVulnerabilities":[{"id":"CVE-2023-46136","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-46136","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pallets/werkzeug/commit/f3c803b3ade485a45f12b6d6617595350c0f03e2","https://github.com/pallets/werkzeug/security/advisories/GHSA-hrfv-mqp8-q5rw","https://security.netapp.com/advisory/ntap-20231124-0008/","https://github.com/pallets/werkzeug/commit/f3c803b3ade485a45f12b6d6617595350c0f03e2","https://github.com/pallets/werkzeug/security/advisories/GHSA-hrfv-mqp8-q5rw","https://security.netapp.com/advisory/ntap-20231124-0008/"],"description":"Werkzeug is a comprehensive WSGI web application library. If an upload of a file that starts with CR or LF and then is followed by megabytes of data without these characters: all of these bytes are appended chunk by chunk into internal bytearray and lookup for boundary is performed on growing buffer. This allows an attacker to cause a denial of service by sending crafted multipart data to an endpoint that will parse it. The amount of CPU time required can block worker processes from handling legitimate requests. This vulnerability has been patched in version 3.0.1.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:A/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":8,"exploitabilityScore":2.1,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-46136","epss":0.0041,"percentile":0.60519,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"werkzeug","version":"2.0.2"}},"found":{"versionConstraint":"<2.3.8 (python)","vulnerabilityID":"GHSA-hrfv-mqp8-q5rw"},"fix":{"suggestedVersion":"2.3.8"}}],"artifact":{"id":"25eafd70a0c695cf","name":"werkzeug","version":"2.0.2","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/Werkzeug-2.0.2.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/Werkzeug-2.0.2.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/Werkzeug-2.0.2.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/Werkzeug-2.0.2.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:armin_ronacher_project:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python:werkzeug:2.0.2:*:*:*:*:*:*:*"],"purl":"pkg:pypi/werkzeug@2.0.2","upstreams":[]}},{"vulnerability":{"id":"GHSA-cx63-2mw6-8hw5","dataSource":"https://github.com/advisories/GHSA-cx63-2mw6-8hw5","namespace":"github:language:python","severity":"High","urls":[],"description":"setuptools vulnerable to Command Injection via package URL","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.8,"exploitabilityScore":2.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-6345","epss":0.00227,"percentile":0.45716,"date":"2025-06-02"}],"fix":{"versions":["70.0.0"],"state":"fixed"},"advisories":[],"risk":0.185005},"relatedVulnerabilities":[{"id":"CVE-2024-6345","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-6345","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pypa/setuptools/commit/88807c7062788254f654ea8c03427adc859321f0","https://huntr.com/bounties/d6362117-ad57-4e83-951f-b8141c6e7ca5","https://github.com/pypa/setuptools/commit/88807c7062788254f654ea8c03427adc859321f0","https://huntr.com/bounties/d6362117-ad57-4e83-951f-b8141c6e7ca5"],"description":"A vulnerability in the package_index module of pypa/setuptools versions up to 69.1.1 allows for remote code execution via its download functions. These functions, which are used to download packages from URLs provided by users or retrieved from package index servers, are susceptible to code injection. If these functions are exposed to user-controlled inputs, such as package URLs, they can execute arbitrary commands on the system. The issue is fixed in version 70.0.","cvss":[{"source":"security@huntr.dev","type":"Secondary","version":"3.0","vector":"CVSS:3.0/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.8,"exploitabilityScore":2.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-6345","epss":0.00227,"percentile":0.45716,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"setuptools","version":"59.6.0"}},"found":{"versionConstraint":"<70.0.0 (python)","vulnerabilityID":"GHSA-cx63-2mw6-8hw5"},"fix":{"suggestedVersion":"70.0.0"}}],"artifact":{"id":"904f7c7f07290780","name":"setuptools","version":"59.6.0","type":"python","locations":[{"path":"/lib/python3.10/site-packages/setuptools-59.6.0.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/setuptools-59.6.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/setuptools-59.6.0.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/setuptools-59.6.0.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/setuptools-59.6.0.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/setuptools-59.6.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["UNKNOWN"],"cpes":["cpe:2.3:a:python:setuptools:59.6.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/setuptools@59.6.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-m2qf-hxjv-5gpq","dataSource":"https://github.com/advisories/GHSA-m2qf-hxjv-5gpq","namespace":"github:language:python","severity":"High","urls":[],"description":"Flask vulnerable to possible disclosure of permanent session cookie due to missing Vary: Cookie header","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:N","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-30861","epss":0.00199,"percentile":0.42671,"date":"2025-06-02"}],"fix":{"versions":["2.2.5"],"state":"fixed"},"advisories":[],"risk":0.14925},"relatedVulnerabilities":[{"id":"CVE-2023-30861","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-30861","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pallets/flask/commit/70f906c51ce49c485f1d355703e9cc3386b1cc2b","https://github.com/pallets/flask/commit/afd63b16170b7c047f5758eb910c416511e9c965","https://github.com/pallets/flask/releases/tag/2.2.5","https://github.com/pallets/flask/releases/tag/2.3.2","https://github.com/pallets/flask/security/advisories/GHSA-m2qf-hxjv-5gpq","https://lists.debian.org/debian-lts-announce/2023/08/msg00024.html","https://security.netapp.com/advisory/ntap-20230818-0006/","https://www.debian.org/security/2023/dsa-5442","https://github.com/pallets/flask/commit/70f906c51ce49c485f1d355703e9cc3386b1cc2b","https://github.com/pallets/flask/commit/afd63b16170b7c047f5758eb910c416511e9c965","https://github.com/pallets/flask/releases/tag/2.2.5","https://github.com/pallets/flask/releases/tag/2.3.2","https://github.com/pallets/flask/security/advisories/GHSA-m2qf-hxjv-5gpq","https://lists.debian.org/debian-lts-announce/2023/08/msg00024.html","https://security.netapp.com/advisory/ntap-20230818-0006/","https://www.debian.org/security/2023/dsa-5442"],"description":"Flask is a lightweight WSGI web application framework. When all of the following conditions are met, a response containing data intended for one client may be cached and subsequently sent by the proxy to other clients. If the proxy also caches `Set-Cookie` headers, it may send one client's `session` cookie to other clients. The severity depends on the application's use of the session and the proxy's behavior regarding cookies. The risk depends on all these conditions being met.\n\n1. The application must be hosted behind a caching proxy that does not strip cookies or ignore responses with cookies.\n2. The application sets `session.permanent = True`\n3. The application does not access or modify the session at any point during a request.\n4. `SESSION_REFRESH_EACH_REQUEST` enabled (the default).\n5. The application does not set a `Cache-Control` header to indicate that a page is private or should not be cached.\n\nThis happens because vulnerable versions of Flask only set the `Vary: Cookie` header when the session is accessed or modified, not when it is refreshed (re-sent to update the expiration) without being accessed or modified. This issue has been fixed in versions 2.3.2 and 2.2.5.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:N","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:N","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-30861","epss":0.00199,"percentile":0.42671,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"flask","version":"2.0.1"}},"found":{"versionConstraint":"<2.2.5 (python)","vulnerabilityID":"GHSA-m2qf-hxjv-5gpq"},"fix":{"suggestedVersion":"2.2.5"}}],"artifact":{"id":"033bdd1b24dbb351","name":"flask","version":"2.0.1","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/Flask-2.0.1.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/Flask-2.0.1.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/Flask-2.0.1.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/Flask-2.0.1.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:armin_ronacher_project:python-flask:2.0.1:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:python_flask:2.0.1:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python-flask:2.0.1:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python_flask:2.0.1:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:flask:2.0.1:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python-flask:2.0.1:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python_flask:2.0.1:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python-flask:2.0.1:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python_flask:2.0.1:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:flask:2.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python-flask:python-flask:2.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python-flask:python_flask:2.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python_flask:python-flask:2.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python_flask:python_flask:2.0.1:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:flask:2.0.1:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:flask:2.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python-flask:2.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python_flask:2.0.1:*:*:*:*:*:*:*","cpe:2.3:a:flask:python-flask:2.0.1:*:*:*:*:*:*:*","cpe:2.3:a:flask:python_flask:2.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python-flask:flask:2.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python_flask:flask:2.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python:flask:2.0.1:*:*:*:*:*:*:*","cpe:2.3:a:flask:flask:2.0.1:*:*:*:*:*:*:*"],"purl":"pkg:pypi/flask@2.0.1","upstreams":[]}},{"vulnerability":{"id":"GHSA-xqr8-7jwr-rhp7","dataSource":"https://github.com/advisories/GHSA-xqr8-7jwr-rhp7","namespace":"github:language:python","severity":"High","urls":[],"description":"Removal of e-Tugra root certificate","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:N","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-37920","epss":0.00112,"percentile":0.30834,"date":"2025-06-02"}],"fix":{"versions":["2023.7.22"],"state":"fixed"},"advisories":[],"risk":0.08399999999999999},"relatedVulnerabilities":[{"id":"CVE-2023-37920","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-37920","namespace":"nvd:cpe","severity":"Critical","urls":["https://github.com/certifi/python-certifi/commit/8fb96ed81f71e7097ed11bc4d9b19afd7ea5c909","https://github.com/certifi/python-certifi/security/advisories/GHSA-xqr8-7jwr-rhp7","https://groups.google.com/a/mozilla.org/g/dev-security-policy/c/C-HrP1SEq1A","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5EX6NG7WUFNUKGFHLM35KHHU3GAKXRTG/","https://github.com/certifi/python-certifi/commit/8fb96ed81f71e7097ed11bc4d9b19afd7ea5c909","https://github.com/certifi/python-certifi/security/advisories/GHSA-xqr8-7jwr-rhp7","https://groups.google.com/a/mozilla.org/g/dev-security-policy/c/C-HrP1SEq1A","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5EX6NG7WUFNUKGFHLM35KHHU3GAKXRTG/","https://security.netapp.com/advisory/ntap-20240912-0002/"],"description":"Certifi is a curated collection of Root Certificates for validating the trustworthiness of SSL certificates while verifying the identity of TLS hosts. Certifi prior to version 2023.07.22 recognizes \"e-Tugra\" root certificates. e-Tugra's root certificates were subject to an investigation prompted by reporting of security issues in their systems. Certifi 2023.07.22 removes root certificates from \"e-Tugra\" from the root store.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":9.8,"exploitabilityScore":3.9,"impactScore":5.9},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:N","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-37920","epss":0.00112,"percentile":0.30834,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"certifi","version":"2020.6.20"}},"found":{"versionConstraint":">=2015.4.28,<2023.7.22 (python)","vulnerabilityID":"GHSA-xqr8-7jwr-rhp7"},"fix":{"suggestedVersion":"2023.7.22"}}],"artifact":{"id":"fcb59256428582cf","name":"certifi","version":"2020.6.20","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/certifi-2020.6.20.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/certifi-2020.6.20.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/certifi-2020.6.20.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/certifi-2020.6.20.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MPL-2.0"],"cpes":["cpe:2.3:a:certifi:certifi:2020.6.20:*:*:*:*:python:*:*"],"purl":"pkg:pypi/certifi@2020.6.20","upstreams":[]}},{"vulnerability":{"id":"GHSA-q2x7-8rv6-6q7h","dataSource":"https://github.com/advisories/GHSA-q2x7-8rv6-6q7h","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Jinja has a sandbox breakout through indirect reference to format method","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":7.8,"exploitabilityScore":1.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-56326","epss":0.00103,"percentile":0.29451,"date":"2025-06-02"}],"fix":{"versions":["3.1.5"],"state":"fixed"},"advisories":[],"risk":0.06592},"relatedVulnerabilities":[{"id":"CVE-2024-56326","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-56326","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/pallets/jinja/commit/48b0687e05a5466a91cd5812d604fa37ad0943b4","https://github.com/pallets/jinja/releases/tag/3.1.5","https://github.com/pallets/jinja/security/advisories/GHSA-q2x7-8rv6-6q7h"],"description":"Jinja is an extensible templating engine. Prior to 3.1.5, An oversight in how the Jinja sandboxed environment detects calls to str.format allows an attacker that controls the content of a template to execute arbitrary Python code. To exploit the vulnerability, an attacker needs to control the content of a template. Whether that is the case depends on the type of application using Jinja. This vulnerability impacts users of applications which execute untrusted templates. Jinja's sandbox does catch calls to str.format and ensures they don't escape the sandbox. However, it's possible to store a reference to a malicious string's format method, then pass that to a filter that calls it. No such filters are built-in to Jinja, but could be present through custom filters in an application. After the fix, such indirect calls are also handled by the sandbox. This vulnerability is fixed in 3.1.5.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:L/AC:L/AT:P/PR:L/UI:P/VC:H/VI:H/VA:H/SC:N/SI:N/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":5.4},"vendorMetadata":{}},{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":7.8,"exploitabilityScore":1.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-56326","epss":0.00103,"percentile":0.29451,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"jinja2","version":"3.0.3"}},"found":{"versionConstraint":"<=3.1.4 (python)","vulnerabilityID":"GHSA-q2x7-8rv6-6q7h"},"fix":{"suggestedVersion":"3.1.5"}}],"artifact":{"id":"f01898c4feedde11","name":"jinja2","version":"3.0.3","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/Jinja2-3.0.3.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/Jinja2-3.0.3.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/Jinja2-3.0.3.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/Jinja2-3.0.3.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:armin_ronacher_project:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python:jinja2:3.0.3:*:*:*:*:*:*:*"],"purl":"pkg:pypi/jinja2@3.0.3","upstreams":[]}},{"vulnerability":{"id":"GHSA-px8h-6qxv-m22q","dataSource":"https://github.com/advisories/GHSA-px8h-6qxv-m22q","namespace":"github:language:python","severity":"Low","urls":[],"description":"Incorrect parsing of nameless cookies leads to __Host- cookies bypass","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:A/AC:H/PR:N/UI:R/S:U/C:N/I:L/A:N","metrics":{"baseScore":2.6,"exploitabilityScore":1.2,"impactScore":1.5},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-23934","epss":0.00222,"percentile":0.45176,"date":"2025-06-02"}],"fix":{"versions":["2.2.3"],"state":"fixed"},"advisories":[],"risk":0.062160000000000014},"relatedVulnerabilities":[{"id":"CVE-2023-23934","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-23934","namespace":"nvd:cpe","severity":"Low","urls":["https://github.com/pallets/werkzeug/commit/cf275f42acad1b5950c50ffe8ef58fe62cdce028","https://github.com/pallets/werkzeug/releases/tag/2.2.3","https://github.com/pallets/werkzeug/security/advisories/GHSA-px8h-6qxv-m22q","https://security.netapp.com/advisory/ntap-20230818-0003/","https://www.debian.org/security/2023/dsa-5470","https://github.com/pallets/werkzeug/commit/cf275f42acad1b5950c50ffe8ef58fe62cdce028","https://github.com/pallets/werkzeug/releases/tag/2.2.3","https://github.com/pallets/werkzeug/security/advisories/GHSA-px8h-6qxv-m22q","https://security.netapp.com/advisory/ntap-20230818-0003/","https://www.debian.org/security/2023/dsa-5470"],"description":"Werkzeug is a comprehensive WSGI web application library. Browsers may allow \"nameless\" cookies that look like `=value` instead of `key=value`. A vulnerable browser may allow a compromised application on an adjacent subdomain to exploit this to set a cookie like `=__Host-test=bad` for another subdomain. Werkzeug prior to 2.2.3 will parse the cookie `=__Host-test=bad` as __Host-test=bad`. If a Werkzeug application is running next to a vulnerable or malicious subdomain which sets such a cookie using a vulnerable browser, the Werkzeug application will see the bad cookie value but the valid cookie key. The issue is fixed in Werkzeug 2.2.3.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:A/AC:L/PR:N/UI:R/S:U/C:N/I:L/A:N","metrics":{"baseScore":3.5,"exploitabilityScore":2.1,"impactScore":1.5},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:A/AC:H/PR:N/UI:R/S:U/C:N/I:L/A:N","metrics":{"baseScore":2.6,"exploitabilityScore":1.2,"impactScore":1.5},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-23934","epss":0.00222,"percentile":0.45176,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"werkzeug","version":"2.0.2"}},"found":{"versionConstraint":"<2.2.3 (python)","vulnerabilityID":"GHSA-px8h-6qxv-m22q"},"fix":{"suggestedVersion":"2.2.3"}}],"artifact":{"id":"25eafd70a0c695cf","name":"werkzeug","version":"2.0.2","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/Werkzeug-2.0.2.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/Werkzeug-2.0.2.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/Werkzeug-2.0.2.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/Werkzeug-2.0.2.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:armin_ronacher_project:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python:werkzeug:2.0.2:*:*:*:*:*:*:*"],"purl":"pkg:pypi/werkzeug@2.0.2","upstreams":[]}},{"vulnerability":{"id":"GHSA-h5c8-rqwp-cp95","dataSource":"https://github.com/advisories/GHSA-h5c8-rqwp-cp95","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Jinja vulnerable to HTML attribute injection when passing user input as keys to xmlattr filter","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:L/I:L/A:N","metrics":{"baseScore":5.4,"exploitabilityScore":2.9,"impactScore":2.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-22195","epss":0.00102,"percentile":0.29274,"date":"2025-06-02"}],"fix":{"versions":["3.1.3"],"state":"fixed"},"advisories":[],"risk":0.05304000000000001},"relatedVulnerabilities":[{"id":"CVE-2024-22195","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-22195","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/pallets/jinja/releases/tag/3.1.3","https://github.com/pallets/jinja/security/advisories/GHSA-h5c8-rqwp-cp95","https://lists.debian.org/debian-lts-announce/2024/01/msg00010.html","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5XCWZD464AJJJUBOO7CMPXQ4ROBC6JX2/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/DELCVUUYX75I5K4Q5WMJG4MUZJA6VAIP/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/O7YWRBX6JQCWC2XXCTZ55C7DPMGICCN3/","https://github.com/pallets/jinja/releases/tag/3.1.3","https://github.com/pallets/jinja/security/advisories/GHSA-h5c8-rqwp-cp95","https://lists.debian.org/debian-lts-announce/2024/01/msg00010.html","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5XCWZD464AJJJUBOO7CMPXQ4ROBC6JX2/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/DELCVUUYX75I5K4Q5WMJG4MUZJA6VAIP/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/O7YWRBX6JQCWC2XXCTZ55C7DPMGICCN3/"],"description":"Jinja is an extensible templating engine. Special placeholders in the template allow writing code similar to Python syntax. It is possible to inject arbitrary HTML attributes into the rendered HTML template, potentially leading to Cross-Site Scripting (XSS). The Jinja `xmlattr` filter can be abused to inject arbitrary HTML attribute keys and values, bypassing the auto escaping mechanism and potentially leading to XSS. It may also be possible to bypass attribute validation checks if they are blacklist-based.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:C/C:L/I:L/A:N","metrics":{"baseScore":6.1,"exploitabilityScore":2.9,"impactScore":2.8},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:L/I:L/A:N","metrics":{"baseScore":5.4,"exploitabilityScore":2.9,"impactScore":2.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-22195","epss":0.00102,"percentile":0.29274,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"jinja2","version":"3.0.3"}},"found":{"versionConstraint":"<3.1.3 (python)","vulnerabilityID":"GHSA-h5c8-rqwp-cp95"},"fix":{"suggestedVersion":"3.1.3"}}],"artifact":{"id":"f01898c4feedde11","name":"jinja2","version":"3.0.3","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/Jinja2-3.0.3.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/Jinja2-3.0.3.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/Jinja2-3.0.3.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/Jinja2-3.0.3.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:armin_ronacher_project:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python:jinja2:3.0.3:*:*:*:*:*:*:*"],"purl":"pkg:pypi/jinja2@3.0.3","upstreams":[]}},{"vulnerability":{"id":"GHSA-h75v-3vvj-5mfj","dataSource":"https://github.com/advisories/GHSA-h75v-3vvj-5mfj","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Jinja vulnerable to HTML attribute injection when passing user input as keys to xmlattr filter","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:L/I:L/A:N","metrics":{"baseScore":5.4,"exploitabilityScore":2.9,"impactScore":2.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-34064","epss":0.00085,"percentile":0.25888,"date":"2025-06-02"}],"fix":{"versions":["3.1.4"],"state":"fixed"},"advisories":[],"risk":0.0442},"relatedVulnerabilities":[{"id":"CVE-2024-34064","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-34064","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/pallets/jinja/commit/0668239dc6b44ef38e7a6c9f91f312fd4ca581cb","https://github.com/pallets/jinja/security/advisories/GHSA-h75v-3vvj-5mfj","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/567XIGSZMABG6TSMYWD7MIYNJSUQQRUC/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/GCLF44KY43BSVMTE6S53B4V5WP3FRRSE/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/SSCBHIL6BYKR5NRCBXP4XMP2CEEKGFVS/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/ZALNWE3TXPPHVPSI3AZ5CTMSTAVN5UMS/","https://github.com/pallets/jinja/commit/0668239dc6b44ef38e7a6c9f91f312fd4ca581cb","https://github.com/pallets/jinja/security/advisories/GHSA-h75v-3vvj-5mfj","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/567XIGSZMABG6TSMYWD7MIYNJSUQQRUC/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/GCLF44KY43BSVMTE6S53B4V5WP3FRRSE/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/SSCBHIL6BYKR5NRCBXP4XMP2CEEKGFVS/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/ZALNWE3TXPPHVPSI3AZ5CTMSTAVN5UMS/"],"description":"Jinja is an extensible templating engine. The `xmlattr` filter in affected versions of Jinja accepts keys containing non-attribute characters. XML/HTML attributes cannot contain spaces, `/`, `>`, or `=`, as each would then be interpreted as starting a separate attribute. If an application accepts keys (as opposed to only values) as user input, and renders these in pages that other users see as well, an attacker could use this to inject other attributes and perform XSS. The fix for CVE-2024-22195 only addressed spaces but not other characters. Accepting keys as user input is now explicitly considered an unintended use case of the `xmlattr` filter, and code that does so without otherwise validating the input should be flagged as insecure, regardless of Jinja version. Accepting _values_ as user input continues to be safe. This vulnerability is fixed in 3.1.4.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:L/I:L/A:N","metrics":{"baseScore":5.4,"exploitabilityScore":2.9,"impactScore":2.6},"vendorMetadata":{}},{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:L/I:L/A:N","metrics":{"baseScore":5.4,"exploitabilityScore":2.9,"impactScore":2.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-34064","epss":0.00085,"percentile":0.25888,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"jinja2","version":"3.0.3"}},"found":{"versionConstraint":"<3.1.4 (python)","vulnerabilityID":"GHSA-h75v-3vvj-5mfj"},"fix":{"suggestedVersion":"3.1.4"}}],"artifact":{"id":"f01898c4feedde11","name":"jinja2","version":"3.0.3","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/Jinja2-3.0.3.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/Jinja2-3.0.3.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/Jinja2-3.0.3.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/Jinja2-3.0.3.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:armin_ronacher_project:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python:jinja2:3.0.3:*:*:*:*:*:*:*"],"purl":"pkg:pypi/jinja2@3.0.3","upstreams":[]}},{"vulnerability":{"id":"GHSA-gmj6-6f8f-6699","dataSource":"https://github.com/advisories/GHSA-gmj6-6f8f-6699","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Jinja has a sandbox breakout through malicious filenames","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.8,"exploitabilityScore":2.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-56201","epss":0.00064,"percentile":0.20401,"date":"2025-06-02"}],"fix":{"versions":["3.1.5"],"state":"fixed"},"advisories":[],"risk":0.044160000000000005},"relatedVulnerabilities":[{"id":"CVE-2024-56201","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-56201","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/pallets/jinja/commit/767b23617628419ae3709ccfb02f9602ae9fe51f","https://github.com/pallets/jinja/issues/1792","https://github.com/pallets/jinja/releases/tag/3.1.5","https://github.com/pallets/jinja/security/advisories/GHSA-gmj6-6f8f-6699"],"description":"Jinja is an extensible templating engine. In versions on the 3.x branch prior to 3.1.5, a bug in the Jinja compiler allows an attacker that controls both the content and filename of a template to execute arbitrary Python code, regardless of if Jinja's sandbox is used. To exploit the vulnerability, an attacker needs to control both the filename and the contents of a template. Whether that is the case depends on the type of application using Jinja. This vulnerability impacts users of applications which execute untrusted templates where the template author can also choose the template filename. This vulnerability is fixed in 3.1.5.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:L/AC:L/AT:P/PR:L/UI:P/VC:H/VI:H/VA:H/SC:N/SI:N/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":5.4},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-56201","epss":0.00064,"percentile":0.20401,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"jinja2","version":"3.0.3"}},"found":{"versionConstraint":">=3.0.0,<=3.1.4 (python)","vulnerabilityID":"GHSA-gmj6-6f8f-6699"},"fix":{"suggestedVersion":"3.1.5"}}],"artifact":{"id":"f01898c4feedde11","name":"jinja2","version":"3.0.3","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/Jinja2-3.0.3.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/Jinja2-3.0.3.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/Jinja2-3.0.3.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/Jinja2-3.0.3.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:armin_ronacher_project:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python:jinja2:3.0.3:*:*:*:*:*:*:*"],"purl":"pkg:pypi/jinja2@3.0.3","upstreams":[]}},{"vulnerability":{"id":"GHSA-2g68-c3qc-8985","dataSource":"https://github.com/advisories/GHSA-2g68-c3qc-8985","namespace":"github:language:python","severity":"High","urls":[],"description":"Werkzeug debugger vulnerable to remote execution when interacting with attacker controlled domain","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":1.7,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-34069","epss":0.0005,"percentile":0.15341,"date":"2025-06-02"}],"fix":{"versions":["3.0.3"],"state":"fixed"},"advisories":[],"risk":0.0375},"relatedVulnerabilities":[{"id":"CVE-2024-34069","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-34069","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pallets/werkzeug/commit/3386395b24c7371db11a5b8eaac0c91da5362692","https://github.com/pallets/werkzeug/security/advisories/GHSA-2g68-c3qc-8985","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/H4SH32AM3CTPMAAEOIDAN7VU565LO4IR/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/HFERFN7PINV4MOGMGA3DPIXJPDCYOEJZ/","https://security.netapp.com/advisory/ntap-20240614-0004/","https://github.com/pallets/werkzeug/commit/3386395b24c7371db11a5b8eaac0c91da5362692","https://github.com/pallets/werkzeug/security/advisories/GHSA-2g68-c3qc-8985","https://lists.debian.org/debian-lts-announce/2025/02/msg00026.html","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/H4SH32AM3CTPMAAEOIDAN7VU565LO4IR/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/HFERFN7PINV4MOGMGA3DPIXJPDCYOEJZ/","https://security.netapp.com/advisory/ntap-20240614-0004/"],"description":"Werkzeug is a comprehensive WSGI web application library. The debugger in affected versions of Werkzeug can allow an attacker to execute code on a developer's machine under some circumstances. This requires the attacker to get the developer to interact with a domain and subdomain they control, and enter the debugger PIN, but if they are successful it allows access to the debugger even if it is only running on localhost. This also requires the attacker to guess a URL in the developer's application that will trigger the debugger. This vulnerability is fixed in 3.0.3.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":1.7,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-34069","epss":0.0005,"percentile":0.15341,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"werkzeug","version":"2.0.2"}},"found":{"versionConstraint":"<3.0.3 (python)","vulnerabilityID":"GHSA-2g68-c3qc-8985"},"fix":{"suggestedVersion":"3.0.3"}}],"artifact":{"id":"25eafd70a0c695cf","name":"werkzeug","version":"2.0.2","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/Werkzeug-2.0.2.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/Werkzeug-2.0.2.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/Werkzeug-2.0.2.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/Werkzeug-2.0.2.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:armin_ronacher_project:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python:werkzeug:2.0.2:*:*:*:*:*:*:*"],"purl":"pkg:pypi/werkzeug@2.0.2","upstreams":[]}},{"vulnerability":{"id":"GHSA-g4mx-q9vg-27p4","dataSource":"https://github.com/advisories/GHSA-g4mx-q9vg-27p4","namespace":"github:language:python","severity":"Medium","urls":[],"description":"urllib3's request body not stripped after redirect from 303 status changes request method to GET","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:A/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N","metrics":{"baseScore":4.2,"exploitabilityScore":0.6,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-45803","epss":0.00062,"percentile":0.19668,"date":"2025-06-02"}],"fix":{"versions":["1.26.18"],"state":"fixed"},"advisories":[],"risk":0.02852},"relatedVulnerabilities":[{"id":"CVE-2023-45803","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-45803","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/urllib3/urllib3/commit/4e98d57809dacab1cbe625fddeec1a290c478ea9","https://github.com/urllib3/urllib3/security/advisories/GHSA-g4mx-q9vg-27p4","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/4R2Y5XK3WALSR3FNAGN7JBYV2B343ZKB/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5F5CUBAN5XMEBVBZPHFITBLMJV5FIJJ5/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/PPDPLM6UUMN55ESPQWJFLLIZY4ZKCNRX/","https://www.rfc-editor.org/rfc/rfc9110.html#name-get","https://github.com/urllib3/urllib3/commit/4e98d57809dacab1cbe625fddeec1a290c478ea9","https://github.com/urllib3/urllib3/security/advisories/GHSA-g4mx-q9vg-27p4","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/4R2Y5XK3WALSR3FNAGN7JBYV2B343ZKB/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5F5CUBAN5XMEBVBZPHFITBLMJV5FIJJ5/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/PPDPLM6UUMN55ESPQWJFLLIZY4ZKCNRX/","https://www.rfc-editor.org/rfc/rfc9110.html#name-get"],"description":"urllib3 is a user-friendly HTTP client library for Python. urllib3 previously wouldn't remove the HTTP request body when an HTTP redirect response using status 301, 302, or 303 after the request had its method changed from one that could accept a request body (like `POST`) to `GET` as is required by HTTP RFCs. Although this behavior is not specified in the section for redirects, it can be inferred by piecing together information from different sections and we have observed the behavior in other major HTTP client implementations like curl and web browsers. Because the vulnerability requires a previously trusted service to become compromised in order to have an impact on confidentiality we believe the exploitability of this vulnerability is low. Additionally, many users aren't putting sensitive data in HTTP request bodies, if this is the case then this vulnerability isn't exploitable. Both of the following conditions must be true to be affected by this vulnerability: 1. Using urllib3 and submitting sensitive information in the HTTP request body (such as form data or JSON) and 2. The origin service is compromised and starts redirecting using 301, 302, or 303 to a malicious peer or the redirected-to service becomes compromised. This issue has been addressed in versions 1.26.18 and 2.0.7 and users are advised to update to resolve this issue. Users unable to update should disable redirects for services that aren't expecting to respond with redirects with `redirects=False` and disable automatic redirects with `redirects=False` and handle 301, 302, and 303 redirects manually by stripping the HTTP request body.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:A/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N","metrics":{"baseScore":4.2,"exploitabilityScore":0.6,"impactScore":3.6},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:A/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N","metrics":{"baseScore":4.2,"exploitabilityScore":0.6,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-45803","epss":0.00062,"percentile":0.19668,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"urllib3","version":"1.26.5"}},"found":{"versionConstraint":">=0,<1.26.18 (python)","vulnerabilityID":"GHSA-g4mx-q9vg-27p4"},"fix":{"suggestedVersion":"1.26.18"}}],"artifact":{"id":"043d634961733646","name":"urllib3","version":"1.26.5","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/urllib3-1.26.5.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/urllib3-1.26.5.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/urllib3-1.26.5.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/urllib3-1.26.5.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:python:urllib3:1.26.5:*:*:*:*:*:*:*"],"purl":"pkg:pypi/urllib3@1.26.5","upstreams":[]}},{"vulnerability":{"id":"GHSA-43fp-rhv2-5gv8","dataSource":"https://github.com/advisories/GHSA-43fp-rhv2-5gv8","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Certifi removing TrustCor root certificate","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:H/UI:N/S:C/C:N/I:H/A:N","metrics":{"baseScore":6.8,"exploitabilityScore":2.3,"impactScore":4},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-23491","epss":0.00038,"percentile":0.10395,"date":"2025-06-02"}],"fix":{"versions":["2022.12.07"],"state":"fixed"},"advisories":[],"risk":0.02242},"relatedVulnerabilities":[{"id":"CVE-2022-23491","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2022-23491","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/certifi/python-certifi/security/advisories/GHSA-43fp-rhv2-5gv8","https://groups.google.com/a/mozilla.org/g/dev-security-policy/c/oxX69KFvsm4/m/yLohoVqtCgAJ","https://github.com/certifi/python-certifi/security/advisories/GHSA-43fp-rhv2-5gv8","https://groups.google.com/a/mozilla.org/g/dev-security-policy/c/oxX69KFvsm4/m/yLohoVqtCgAJ","https://security.netapp.com/advisory/ntap-20230223-0010/"],"description":"Certifi is a curated collection of Root Certificates for validating the trustworthiness of SSL certificates while verifying the identity of TLS hosts. Certifi 2022.12.07 removes root certificates from \"TrustCor\" from the root store. These are in the process of being removed from Mozilla's trust store. TrustCor's root certificates are being removed pursuant to an investigation prompted by media reporting that TrustCor's ownership also operated a business that produced spyware. Conclusions of Mozilla's investigation can be found in the linked google group discussion.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:N","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:H/UI:N/S:C/C:N/I:H/A:N","metrics":{"baseScore":6.8,"exploitabilityScore":2.3,"impactScore":4},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-23491","epss":0.00038,"percentile":0.10395,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"certifi","version":"2020.6.20"}},"found":{"versionConstraint":">=2017.11.05,<2022.12.07 (python)","vulnerabilityID":"GHSA-43fp-rhv2-5gv8"},"fix":{"suggestedVersion":"2022.12.07"}}],"artifact":{"id":"fcb59256428582cf","name":"certifi","version":"2020.6.20","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/certifi-2020.6.20.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/certifi-2020.6.20.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/certifi-2020.6.20.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/certifi-2020.6.20.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MPL-2.0"],"cpes":["cpe:2.3:a:certifi:certifi:2020.6.20:*:*:*:*:python:*:*"],"purl":"pkg:pypi/certifi@2020.6.20","upstreams":[]}},{"vulnerability":{"id":"GHSA-mq26-g339-26xf","dataSource":"https://github.com/advisories/GHSA-mq26-g339-26xf","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Command Injection in pip when used with Mercurial","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:H/A:N","metrics":{"baseScore":5.5,"exploitabilityScore":1.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-5752","epss":0.0004,"percentile":0.11732,"date":"2025-06-02"}],"fix":{"versions":["23.3"],"state":"fixed"},"advisories":[],"risk":0.021},"relatedVulnerabilities":[{"id":"CVE-2023-5752","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-5752","namespace":"nvd:cpe","severity":"Low","urls":["https://github.com/pypa/pip/pull/12306","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/622OZXWG72ISQPLM5Y57YCVIMWHD4C3U/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/65UKKF5LBHEFDCUSPBHUN4IHYX7SRMHH/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FXUVMJM25PUAZRQZBF54OFVKTY3MINPW/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KFC2SPFG5FLCZBYY2K3T5MFW2D22NG6E/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YBSB3SUPQ3VIFYUMHPO3MEQI4BJAXKCZ/","https://mail.python.org/archives/list/security-announce@python.org/thread/F4PL35U6X4VVHZ5ILJU3PWUWN7H7LZXL/","https://github.com/pypa/pip/pull/12306","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/622OZXWG72ISQPLM5Y57YCVIMWHD4C3U/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/65UKKF5LBHEFDCUSPBHUN4IHYX7SRMHH/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FXUVMJM25PUAZRQZBF54OFVKTY3MINPW/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KFC2SPFG5FLCZBYY2K3T5MFW2D22NG6E/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YBSB3SUPQ3VIFYUMHPO3MEQI4BJAXKCZ/","https://mail.python.org/archives/list/security-announce@python.org/thread/F4PL35U6X4VVHZ5ILJU3PWUWN7H7LZXL/"],"description":"When installing a package from a Mercurial VCS URL  (ie \"pip install \nhg+...\") with pip prior to v23.3, the specified Mercurial revision could\n be used to inject arbitrary configuration options to the \"hg clone\" \ncall (ie \"--config\"). Controlling the Mercurial configuration can modify\n how and which repository is installed. This vulnerability does not \naffect users who aren't installing from Mercurial.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:L/A:N","metrics":{"baseScore":3.3,"exploitabilityScore":1.9,"impactScore":1.5},"vendorMetadata":{}},{"source":"cna@python.org","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:H/A:N","metrics":{"baseScore":5.5,"exploitabilityScore":1.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-5752","epss":0.0004,"percentile":0.11732,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pip","version":"22.0.2"}},"found":{"versionConstraint":"<23.3 (python)","vulnerabilityID":"GHSA-mq26-g339-26xf"},"fix":{"suggestedVersion":"23.3"}}],"artifact":{"id":"5976e4c20bc29a3b","name":"pip","version":"22.0.2","type":"python","locations":[{"path":"/lib/python3.10/site-packages/pip-22.0.2.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/pip-22.0.2.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/pip-22.0.2.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/pip-22.0.2.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/pip-22.0.2.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/pip-22.0.2.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:pip_developers_project:python-pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers_project:python_pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:distutils_sig_project:python-pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:distutils_sig_project:python_pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip_developersproject:python-pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip_developersproject:python_pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:distutils_sigproject:python-pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:distutils_sigproject:python_pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers_project:pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:distutils_sig_project:pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers:python-pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers:python_pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip_developersproject:pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:distutils-sig:python-pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:distutils-sig:python_pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:distutils_sig:python-pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:distutils_sig:python_pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:distutils_sigproject:pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python-pip:python-pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python-pip:python_pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python_pip:python-pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python_pip:python_pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers:pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:distutils-sig:pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:distutils_sig:pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa:python-pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa:python_pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip:python-pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip:python_pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python-pip:pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python_pip:pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python:pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pypa:pip:22.0.2:*:*:*:*:*:*:*","cpe:2.3:a:pip:pip:22.0.2:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pip@22.0.2","upstreams":[]}},{"vulnerability":{"id":"GHSA-f9vj-2wh5-fj8j","dataSource":"https://github.com/advisories/GHSA-f9vj-2wh5-fj8j","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Werkzeug safe_join not safe on Windows","cvss":[],"epss":[{"cve":"CVE-2024-49766","epss":0.00036,"percentile":0.09088,"date":"2025-06-02"}],"fix":{"versions":["3.0.6"],"state":"fixed"},"advisories":[],"risk":0.018000000000000002},"relatedVulnerabilities":[{"id":"CVE-2024-49766","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-49766","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/pallets/werkzeug/commit/2767bcb10a7dd1c297d812cc5e6d11a474c1f092","https://github.com/pallets/werkzeug/releases/tag/3.0.6","https://github.com/pallets/werkzeug/security/advisories/GHSA-f9vj-2wh5-fj8j","https://security.netapp.com/advisory/ntap-20250131-0005/"],"description":"Werkzeug is a Web Server Gateway Interface web application library. On Python < 3.11 on Windows, os.path.isabs() does not catch UNC paths like //server/share. Werkzeug's safe_join() relies on this check, and so can produce a path that is not safe, potentially allowing unintended access to data. Applications using Python >= 3.11, or not using Windows, are not vulnerable. Werkzeug version 3.0.6 contains a patch.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:N/AC:H/AT:N/PR:N/UI:N/VC:L/VI:N/VA:N/SC:N/SI:N/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":6.3},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-49766","epss":0.00036,"percentile":0.09088,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"werkzeug","version":"2.0.2"}},"found":{"versionConstraint":"<=3.0.5 (python)","vulnerabilityID":"GHSA-f9vj-2wh5-fj8j"},"fix":{"suggestedVersion":"3.0.6"}}],"artifact":{"id":"25eafd70a0c695cf","name":"werkzeug","version":"2.0.2","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/Werkzeug-2.0.2.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/Werkzeug-2.0.2.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/Werkzeug-2.0.2.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/Werkzeug-2.0.2.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:armin_ronacher_project:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python-werkzeug:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python_werkzeug:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python-werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python_werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:werkzeug:werkzeug:2.0.2:*:*:*:*:*:*:*","cpe:2.3:a:python:werkzeug:2.0.2:*:*:*:*:*:*:*"],"purl":"pkg:pypi/werkzeug@2.0.2","upstreams":[]}},{"vulnerability":{"id":"GHSA-cpwx-vrp4-4pq7","dataSource":"https://github.com/advisories/GHSA-cpwx-vrp4-4pq7","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Jinja2 vulnerable to sandbox breakout through attr filter selecting format method","cvss":[],"epss":[{"cve":"CVE-2025-27516","epss":0.00034,"percentile":0.08503,"date":"2025-06-02"}],"fix":{"versions":["3.1.6"],"state":"fixed"},"advisories":[],"risk":0.017},"relatedVulnerabilities":[{"id":"CVE-2025-27516","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2025-27516","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/pallets/jinja/commit/90457bbf33b8662926ae65cdde4c4c32e756e403","https://github.com/pallets/jinja/security/advisories/GHSA-cpwx-vrp4-4pq7","https://lists.debian.org/debian-lts-announce/2025/04/msg00045.html"],"description":"Jinja is an extensible templating engine. Prior to 3.1.6, an oversight in how the Jinja sandboxed environment interacts with the |attr filter allows an attacker that controls the content of a template to execute arbitrary Python code. To exploit the vulnerability, an attacker needs to control the content of a template. Whether that is the case depends on the type of application using Jinja. This vulnerability impacts users of applications which execute untrusted templates. Jinja's sandbox does catch calls to str.format and ensures they don't escape the sandbox. However, it's possible to use the |attr filter to get a reference to a string's plain format method, bypassing the sandbox. After the fix, the |attr filter no longer bypasses the environment's attribute lookup. This vulnerability is fixed in 3.1.6.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:L/AC:L/AT:P/PR:L/UI:P/VC:H/VI:H/VA:H/SC:N/SI:N/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":5.4},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2025-27516","epss":0.00034,"percentile":0.08503,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"jinja2","version":"3.0.3"}},"found":{"versionConstraint":"<=3.1.5 (python)","vulnerabilityID":"GHSA-cpwx-vrp4-4pq7"},"fix":{"suggestedVersion":"3.1.6"}}],"artifact":{"id":"f01898c4feedde11","name":"jinja2","version":"3.0.3","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/Jinja2-3.0.3.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/Jinja2-3.0.3.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/Jinja2-3.0.3.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/Jinja2-3.0.3.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:armin_ronacher_project:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher_project:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacherproject:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin-ronacher:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:armin_ronacher:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python:python-jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python:python_jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:jinja2:3.0.3:*:*:*:*:*:*:*","cpe:2.3:a:python:jinja2:3.0.3:*:*:*:*:*:*:*"],"purl":"pkg:pypi/jinja2@3.0.3","upstreams":[]}},{"vulnerability":{"id":"GHSA-34jh-p97f-mpxf","dataSource":"https://github.com/advisories/GHSA-34jh-p97f-mpxf","namespace":"github:language:python","severity":"Medium","urls":[],"description":"urllib3's Proxy-Authorization request header isn't stripped during cross-origin redirects","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N","metrics":{"baseScore":4.4,"exploitabilityScore":0.8,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-37891","epss":0.00033,"percentile":0.08074,"date":"2025-06-02"}],"fix":{"versions":["1.26.19"],"state":"fixed"},"advisories":[],"risk":0.01551},"relatedVulnerabilities":[{"id":"CVE-2024-37891","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-37891","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/urllib3/urllib3/commit/accff72ecc2f6cf5a76d9570198a93ac7c90270e","https://github.com/urllib3/urllib3/security/advisories/GHSA-34jh-p97f-mpxf","https://github.com/urllib3/urllib3/commit/accff72ecc2f6cf5a76d9570198a93ac7c90270e","https://github.com/urllib3/urllib3/security/advisories/GHSA-34jh-p97f-mpxf","https://security.netapp.com/advisory/ntap-20240822-0003/","https://www.vicarius.io/vsociety/posts/proxy-authorization-header-handling-vulnerability-in-urllib3-cve-2024-37891"],"description":"urllib3 is a user-friendly HTTP client library for Python. When using urllib3's proxy support with `ProxyManager`, the `Proxy-Authorization` header is only sent to the configured proxy, as expected. However, when sending HTTP requests *without* using urllib3's proxy support, it's possible to accidentally configure the `Proxy-Authorization` header even though it won't have any effect as the request is not using a forwarding proxy or a tunneling proxy. In those cases, urllib3 doesn't treat the `Proxy-Authorization` HTTP header as one carrying authentication material and thus doesn't strip the header on cross-origin redirects. Because this is a highly unlikely scenario, we believe the severity of this vulnerability is low for almost all users. Out of an abundance of caution urllib3 will automatically strip the `Proxy-Authorization` header during cross-origin redirects to avoid the small chance that users are doing this on accident. Users should use urllib3's proxy support or disable automatic redirects to achieve safe processing of the `Proxy-Authorization` header, but we still decided to strip the header by default in order to further protect users who aren't using the correct approach. We believe the number of usages affected by this advisory is low. It requires all of the following to be true to be exploited: 1. Setting the `Proxy-Authorization` header without using urllib3's built-in proxy support. 2. Not disabling HTTP redirects. 3. Either not using an HTTPS origin server or for the proxy or target origin to redirect to a malicious origin. Users are advised to update to either version 1.26.19 or version 2.2.2. Users unable to upgrade may use the `Proxy-Authorization` header with urllib3's `ProxyManager`, disable HTTP redirects using `redirects=False` when sending requests, or not user the `Proxy-Authorization` header as mitigations.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N","metrics":{"baseScore":4.4,"exploitabilityScore":0.8,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-37891","epss":0.00033,"percentile":0.08074,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"urllib3","version":"1.26.5"}},"found":{"versionConstraint":"<1.26.19 (python)","vulnerabilityID":"GHSA-34jh-p97f-mpxf"},"fix":{"suggestedVersion":"1.26.19"}}],"artifact":{"id":"043d634961733646","name":"urllib3","version":"1.26.5","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/urllib3-1.26.5.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/urllib3-1.26.5.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/urllib3-1.26.5.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/urllib3-1.26.5.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:python:urllib3:1.26.5:*:*:*:*:*:*:*"],"purl":"pkg:pypi/urllib3@1.26.5","upstreams":[]}},{"vulnerability":{"id":"GHSA-9wx4-h78v-vm56","dataSource":"https://github.com/advisories/GHSA-9wx4-h78v-vm56","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Requests `Session` object does not verify requests after making first request with verify=False","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:H/PR:H/UI:R/S:U/C:H/I:H/A:N","metrics":{"baseScore":5.6,"exploitabilityScore":0.4,"impactScore":5.2},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-35195","epss":0.00009,"percentile":0.0055,"date":"2025-06-02"}],"fix":{"versions":["2.32.0"],"state":"fixed"},"advisories":[],"risk":0.004770000000000001},"relatedVulnerabilities":[{"id":"CVE-2024-35195","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-35195","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/psf/requests/commit/a58d7f2ffb4d00b46dca2d70a3932a0b37e22fac","https://github.com/psf/requests/pull/6655","https://github.com/psf/requests/security/advisories/GHSA-9wx4-h78v-vm56","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/IYLSNK5TL46Q6XPRVMHVWS63MVJQOK4Q/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/N7WP6EYDSUOCOJYHDK5NX43PYZ4SNHGZ/","https://github.com/psf/requests/commit/a58d7f2ffb4d00b46dca2d70a3932a0b37e22fac","https://github.com/psf/requests/pull/6655","https://github.com/psf/requests/security/advisories/GHSA-9wx4-h78v-vm56","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/IYLSNK5TL46Q6XPRVMHVWS63MVJQOK4Q/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/N7WP6EYDSUOCOJYHDK5NX43PYZ4SNHGZ/"],"description":"Requests is a HTTP library. Prior to 2.32.0, when making requests through a Requests `Session`, if the first request is made with `verify=False` to disable cert verification, all subsequent requests to the same host will continue to ignore cert verification regardless of changes to the value of `verify`. This behavior will continue for the lifecycle of the connection in the connection pool. This vulnerability is fixed in 2.32.0.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:H/PR:H/UI:R/S:U/C:H/I:H/A:N","metrics":{"baseScore":5.6,"exploitabilityScore":0.4,"impactScore":5.2},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-35195","epss":0.00009,"percentile":0.0055,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"requests","version":"2.25.1"}},"found":{"versionConstraint":"<2.32.0 (python)","vulnerabilityID":"GHSA-9wx4-h78v-vm56"},"fix":{"suggestedVersion":"2.32.0"}}],"artifact":{"id":"52e214e010c57a74","name":"requests","version":"2.25.1","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/requests-2.25.1.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/requests-2.25.1.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/requests-2.25.1.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/requests-2.25.1.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["Apache 2.0"],"cpes":["cpe:2.3:a:python:requests:2.25.1:*:*:*:*:*:*:*"],"purl":"pkg:pypi/requests@2.25.1","upstreams":[]}}],"source":{"type":"sbom-file","target":"cubiemedia-mqtt-client-syft.json"},"distro":{"name":"","version":"","idLike":null},"descriptor":{"name":"grype","version":"0.92.2","configuration":{"output":["json=/Users/alan/Personal/popey/syft-snap-source-test/grype/v1.26.0/2025-05-21/cubiemedia-mqtt-client-grype.json","template=/Users/alan/Personal/popey/syft-snap-source-test/grype/v1.26.0/2025-05-21/cubiemedia-mqtt-client-grype.html"],"file":"","pretty":false,"distro":"","add-cpes-if-none":false,"output-template-file":"../html.tmpl","check-for-app-update":true,"only-fixed":false,"only-notfixed":false,"ignore-wontfix":"","platform":"","search":{"scope":"squashed","unindexed-archives":false,"indexed-archives":true},"ignore":[{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"kernel-headers","version":"","language":"","type":"rpm","location":"","upstream-name":"kernel"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"},{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"linux(-.*)?-headers-.*","version":"","language":"","type":"deb","location":"","upstream-name":"linux.*"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"},{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"linux-libc-dev","version":"","language":"","type":"deb","location":"","upstream-name":"linux"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"}],"exclude":[],"externalSources":{"enable":false,"maven":{"searchUpstreamBySha1":true,"baseUrl":"https://search.maven.org/solrsearch/select","rateLimit":300000000}},"match":{"java":{"using-cpes":false},"jvm":{"using-cpes":true},"dotnet":{"using-cpes":false},"golang":{"using-cpes":false,"always-use-cpe-for-stdlib":true,"allow-main-module-pseudo-version-comparison":false},"javascript":{"using-cpes":false},"python":{"using-cpes":false},"ruby":{"using-cpes":false},"rust":{"using-cpes":false},"stock":{"using-cpes":true}},"fail-on-severity":"","registry":{"insecure-skip-tls-verify":false,"insecure-use-http":false,"auth":null,"ca-cert":""},"show-suppressed":false,"by-cve":false,"SortBy":{"sort-by":"risk"},"name":"","default-image-pull-source":"","vex-documents":[],"vex-add":[],"match-upstream-kernel-headers":false,"db":{"cache-dir":"/Users/alan/Library/Caches/grype/db","update-url":"https://grype.anchore.io/databases","ca-cert":"","auto-update":true,"validate-by-hash-on-start":true,"validate-age":true,"max-allowed-built-age":432000000000000,"require-update-check":false,"update-available-timeout":30000000000,"update-download-timeout":300000000000,"max-update-check-frequency":7200000000000},"exp":{},"dev":{"db":{"debug":false}}},"db":{"status":{"schemaVersion":"v6.0.2","from":"https://grype.anchore.io/databases/v6/vulnerability-db_v6.0.2_2025-06-03T01:32:03Z_1748923962.tar.zst?checksum=sha256%3A02a6127217518a93e0ef8739cb9f2d659109fff0d9cc25f0c13d5ca8b307a118","built":"2025-06-03T04:12:42Z","path":"/Users/alan/Library/Caches/grype/db/6/vulnerability.db","valid":true},"providers":{"alpine":{"captured":"2025-06-03T01:32:27Z","input":"xxh64:a6a065ece392f6d1"},"amazon":{"captured":"2025-06-03T01:32:10Z","input":"xxh64:b65d458d6858d72a"},"chainguard":{"captured":"2025-06-03T01:32:15Z","input":"xxh64:bad092583f78d4dd"},"debian":{"captured":"2025-06-03T01:32:22Z","input":"xxh64:b4717cad509f50d8"},"epss":{"captured":"2025-06-03T01:32:43Z","input":"xxh64:2f9660be834e0ce6"},"github":{"captured":"2025-06-03T01:32:21Z","input":"xxh64:cdd94f1952fcf607"},"kev":{"captured":"2025-06-03T01:32:03Z","input":"xxh64:575844c3434cc70b"},"mariner":{"captured":"2025-06-03T01:32:11Z","input":"xxh64:7628f1adb0b9b12b"},"nvd":{"captured":"2025-06-03T01:32:30Z","input":"xxh64:463c4e9a1d401101"},"oracle":{"captured":"2025-06-03T01:32:27Z","input":"xxh64:257df3059d28d823"},"rhel":{"captured":"2025-06-03T01:33:04Z","input":"xxh64:59e7d70bf674afdc"},"sles":{"captured":"2025-06-03T01:32:46Z","input":"xxh64:3cf7f372ea0c981a"},"ubuntu":{"captured":"2025-06-03T01:33:20Z","input":"xxh64:075be8afc080f2be"},"wolfi":{"captured":"2025-06-03T01:32:31Z","input":"xxh64:1aa0b26b004bdd38"}}},"timestamp":"2025-06-03T13:02:48.394933+01:00"}}
